src/rules/noDuplicateSuperRule.ts

```diff
@@ -80,6 +80,11 @@ function walk(ctx: Lint.WalkContext<void>): void {
                     ? { node: node.parent! as ts.CallExpression, break: false }
                     : Kind.NoSuper;
 
+            case ts.SyntaxKind.ConditionalExpression: {
+                const { whenTrue, whenFalse } = node as ts.ConditionalExpression;
+                return worse(getSuperForNode(whenTrue), getSuperForNode(whenFalse));
```

Comment:
to also handle `super()` calls inside the condition:
```ts
const inCondition = getSuperForNode(node.condition);
const inBranches = worse(getSuperForNode(whenTrue), getSuperForNode(whenFalse));
if (typeof inCondition !== "number" && typeof inBranches !== number) {
    addDuplicateFailure(inCondition.node, inBranches.node);
}
return worse(inCondition, inBranches);
```

*code is untested and written only in GitHub comment editor.